quic/s2n-quic-transport/src/space/application.rs

```diff
@@ -440,6 +445,22 @@ impl<Config: endpoint::Config> ApplicationSpace<Config> {
         )
     }
 
+    /// Returns `true` if sending is limited by the application and not the congestion controller
+    ///
+    /// Sending is app limited if the application is not fully utilizing the available
+    /// congestion window currently and there is no more application data remaining to send.
+    fn is_app_limited(&self, path: &Path<Config>, bytes_sent: usize) -> bool {
+        let cwnd = path.congestion_controller.congestion_window();
+        let bytes_in_flight = path
+            .congestion_controller
+            .bytes_in_flight()
+            .saturating_add(bytes_sent as u32);
+        let mtu = path.mtu_controller.mtu() as u32;
+
+        let congestion_blocked = cwnd.saturating_sub(bytes_in_flight) < mtu;
+        !congestion_blocked && !self.has_transmission_interest()
```

Comment:
Should we be reporting that we're application blocked if we're congestion blocked? I'm trying to think what that implies...

Reply:
Not sure I understand, if we are congestion blocked then we are not application-blocked